docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,411 @@
+# and distribution of degrees of freedom for our approximated fields.
+# We create the `DofHandler` and then add a single scalar field called `:u` based on
+# our interpolation `ip` defined above.
+# Lastly we `close!` the `DofHandler`, it is now that the dofs are distributed
+# for all the elements.
+dh = DofHandler(grid)
+add!(dh, :u, ip)
+close!(dh);
+
+# Now that we have distributed all our dofs we can create our tangent matrix,
+# using `create_sparsity_pattern`. This function returns a sparse matrix
+# with the correct entries stored. We need to pass the topology and the cross-element coupling matrix when we're using
+# discontinuous interpolations. The cross-element coupling matrix is of size [1,1] in this case as
+# we have only one field and one DofHandler.
+K = create_sparsity_pattern(dh, topology = topology, cross_coupling = trues(1,1))
+
+# ### Boundary conditions
+# In Ferrite constraints like Dirichlet boundary conditions
+# are handled by a `ConstraintHandler`.
+ch = ConstraintHandler(dh);
+
+# Next we need to add constraints to `ch`. For this problem we define
+# inhomogeneous Dirichlet boundary conditions on the two boundaries.
+# We define $\partial \Omega_n$ as the `union` of the face sets with Neumann boundary conditions.
+∂Ωₙ = union(
+    getfaceset(grid, "top"),
+    getfaceset(grid, "bottom"),
+);
+
+# Now we are set up to define our constraint. We specify which field
+# the condition is for, and our face sets. The last
+# argument is a function of the form $f(\textbf{x})$ or $f(\textbf{x}, t)$,
+# where $\textbf{x}$ is the spatial coordinate and
+# $t$ the current time, and returns the prescribed value.
+add!(ch, Dirichlet(:u, getfaceset(grid, "right"), (x, t) -> 1.0));
+add!(ch, Dirichlet(:u, getfaceset(grid, "left"), (x, t) -> -1.0));
+
+# Finally we also need to `close!` our constraint handler. When we call `close!`
+# the dofs corresponding to our constraints are calculated and stored
+# in our `ch` object.
+close!(ch)
+
+# Note that if one or more of the constraints are time dependent we would use
+# [`update!`](@ref) to recompute prescribed values in each new timestep.
+
+# ### Assembling the linear system
+#
+# Now we have all the pieces needed to assemble the linear system, $K u = f$.
+# Assembling of the global system is done by looping over all the elements in order to
+# compute the element contributions ``K_e`` and ``f_e``, all the interfaces
+# to compute their contributions ``K_i``, and all the Neumann boundary faces to compute their
+# contributions ``f_e`` which are then assembled to the
+# appropriate place in the global ``K`` and ``f``.
+#
+# #### Element assembly
+# We define the functions
+# * `assemble_element!` to compute the contribution of volume integrals over an element.
+# * `assemble_interface!` to compute the contribution of surface integrals over an interface.
+# * `assemble_boundary!` to compute the contribution of surface integrals over a boundary face.
```

Comment:
Perhaps this suffices instead of explaining what is used from each object below? And if needed, only explain for the interface values. (The list below of what is used is missing the integration weight)
```suggestion
# * `assemble_element!` to compute the contributions ``K_e`` and ``f_e`` of volume integrals over an element using `cellvalues`.
# * `assemble_interface!` to compute the contribution ``K_i`` of surface integrals over an interface using `interfacevalues`.
# * `assemble_boundary!` to compute the contribution ``f_e`` of surface integrals over a boundary face using `facevalues`.
```